src/session_impl.cpp

```diff
@@ -588,30 +587,7 @@ namespace aux {
 		TORRENT_ASSERT(is_single_thread());
 
 #ifndef TORRENT_DISABLE_LOGGING
-		// this alert is a bit special. The stats headers aren't very useful
```

Comment:
I take it the purpose of this patch is to get rid of the posting of this alert, right?
I really think there's a benefit of having this be an alert that gets posted through the same path as the stats counters. The main reason is that any client that can log alert messages to a file will support being instrumented. There's no extra work on the client's side. I think this is especially useful because (in my experience), being able to instrument and trouble-shoot is something I'm interested in, but not client developers to the same extent. or at least it significantly lowers the hurdle to be able to provide stats.

I'm not sure I have a correct understanding of your concern of having this here. Is it that it creates noise in your alert logs? is it that it blows up the stack allocators by allocating a large string?

What would you think about introducing a separate alert just for this. It would have these benefits:

* It could belong to the stats category and not be mixed up with enabling logging.
* It could be completely stateless and its ``message()`` function could just generate this string (and most of this code would move in there)
* It could easily be identified by a client that wants to ignore it